### PR TITLE
Commit of function encoding_vars

### PR DESCRIPTION
**Pull request by NansK**

### CODE REVIEW

File: scripts/feature_encoder.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+def encoding_vars(dataset):
+    import pandas as pd
+    import numpy as np
+    from sklearn.preprocessing import OrdinalEncoder
+    """
+    This function encode non-numeric features. You pass the OpenFoodFact dataset as entry and you get the encoded dataset at the end.
+    Will be different encoding depending on some factors. Features with ordinal order are encoded with OrdinalEncoder. Features without ',' are encoded with OneHotEncoder. Features with ',' are manually encoded, like a OneHotEncoder would have done, but before we split the datas.
```
Comment (HerySon) [inline scripts/feature_encoder.py:7]: Ligne trop longue, à couper en différentes parties. Peux tu expliciter plus clairement le choix de l'encodeur en fonction des valeurs rencontrées dans les features ?

File: scripts/feature_encoder.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        list_of_cat = columnar.value_counts().index
+        dataset[i] = dataset[i].astype(str)
+        for i in list_of_cat:
+            dataset[i] = np.nan
+            for u in range(len(dataset)):
+                if dataset.loc[u, "allergens"].__contains__(i):
+                    dataset.at[u, i] = 1
+                else:
+                    dataset.at[u, i] = 0
+    dataset.drop(
```
Comment (HerySon) [inline scripts/feature_encoder.py:49]: Peux tu passer en kwarg cette liste ?

File: scripts/feature_encoder.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        categories=[["not-applicable", "unknown", "e", "d", "c", "b", "a"], ["unknown", "e", "d", "c", "b", "a"],
+                    ["unknown", 4.0, 3.0, 2.0, 1.0]])
+    oe.fit(dataset[ordinal_cols])
+    dataset[ordinal_cols] = oe_ecoscore_grade.transform(dataset[ordinal_cols])
+    dataset.drop(ordinal_cols, axis=1, inplace=True)
+    # Dummies (OneHotEncoding)
+    dataset = dataset.join(pd.get_dummies(dataset[dummies_cols]))
+    dataset.drop(dummies_cols, axis=1, inplace=True)
+    # Splits
+    for i in cols_to_split:
```
Comment (HerySon) [inline scripts/feature_encoder.py:33]: Peux tu ajouter des commentaires pour détailler ce qui est fait dans cette boucle ?

**Review by HerySon — CHANGES_REQUESTED** [review]

**Comment by HerySon** [comment]
@NansK 

## Todo :
- [ ] valider les changements mineurs cités plus haut
- [ ] optionnel : ajouter une méthode d'encodage de feature catégorielle alternative à OHE

File: scripts/feature_encoder.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    dummies_cols = ["brands_tags", "origins_fr", "generic_name", "product_name", "first_packaging_code_geo",
+                    "additives_fr", "pnns_groups_1", "pnns_groups_2", "main_category"]
+    # Ordinals (OrdinalEncoder)
+    for i in ordinal_cols:
+        dataset[i].fillna("unknown", inplace=True)
+    oe = OrdinalEncoder(
+        categories=[["not-applicable", "unknown", "e", "d", "c", "b", "a"], ["unknown", "e", "d", "c", "b", "a"],
+                    ["unknown", 4.0, 3.0, 2.0, 1.0]])
+    oe.fit(dataset[ordinal_cols])
+    dataset[ordinal_cols] = oe_ecoscore_grade.transform(dataset[ordinal_cols])
```
[the remark this answers is not in the document]
Comment (NansK) [reply scripts/feature_encoder.py:34]: oui c'est juste oe je vais modifier ça, erreur de copier coller de mes tests du notebook

File: scripts/feature_encoder.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    This function encode non-numeric features. You pass the OpenFoodFact dataset as entry and you get the encoded dataset at the end.
+    Will be different encoding depending on some factors. Features with ordinal order are encoded with OrdinalEncoder. Features without ',' are encoded with OneHotEncoder. Features with ',' are manually encoded, like a OneHotEncoder would have done, but before we split the datas.
+
+    Args :
+        dataset -- the dataset who contains OpenFoodFact data (required)
+
+    Returns :
+        Encoded dataset of OpenFoodFact.
+    """
+    cols_to_split_but_big_memory = ["categories", "ingredients_tags"]
```
[the remark this answers is not in the document]
Comment (NansK) [reply scripts/feature_encoder.py]: Fait

File: scripts/feature_encoder.py (hunk 1 above)
Comment (NansK) [reply scripts/feature_encoder.py]: Fait

File: scripts/feature_encoder.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (NansK) [reply scripts/feature_encoder.py]: Fait

File: scripts/feature_encoder.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    for i in ordinal_cols:
+        dataset[i].fillna("unknown", inplace=True)
+    if oe == True:
+      oe = OrdinalEncoder(
+          categories=[["not-applicable", "unknown", "e", "d", "c", "b", "a"], ["unknown", "e", "d", "c", "b", "a"], ["unknown", 4.0, 3.0, 2.0, 1.0]])
+    else:
+      oe = HashingEncoder()
+    oe.fit(dataset[ordinal_cols])
+    dataset[ordinal_cols] = oe.transform(dataset[ordinal_cols])
+    dataset.drop(ordinal_cols, axis=1, inplace=True)
```
Comment (HerySon) [inline scripts/feature_encoder.py:64]: Je ne comprends pas pourquoi les colonnes `ordinals_cols` sont droppées

File: scripts/feature_encoder.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    # Dummies (OneHotEncoding)
+    dataset = dataset.join(pd.get_dummies(dataset[dummies_cols]))
+    dataset.drop(dummies_cols, axis=1, inplace=True)
```
Comment (HerySon) [inline scripts/feature_encoder.py:67]: Même remarque ici

Comment (NansK) [reply scripts/feature_encoder.py:67]: dataset.drop(dummies_cols, axis=1, inplace=True) est là pour enlever les variables de bases qui étaient présentes dans dataset avant le join, on a obtenu les nouvelles après l'encodage et on a join du coup on se débarrasse des variables initiales.

File: scripts/feature_encoder.py (hunk 6 above)
Comment (NansK) [reply scripts/feature_encoder.py]: C'est enlevé, je me suis expliqué dans le commentaire suivant
